handler.go

```diff
@@ -17,112 +19,169 @@ var (
 		template.New("package.html").Parse(templates.Package))
 )
 
-// CreateHandler creates a Sally http.Handler
+// CreateHandler builds a new handler
+// with the provided package configuration.
+// The returned handler provides the following endpoints:
+//
+//	GET /
+//		Index page listing all packages.
+//	GET /<name>
+//	       Package page for the given package.
+//	GET /<dir>
+//		Page listing packages under the given directory,
+//		assuming that there's no package with the given name.
+//	GET /<name>/<subpkg>
+//		Package page for the given subpackage.
 func CreateHandler(config *Config) http.Handler {
 	mux := http.NewServeMux()
-
-	pkgs := make([]packageInfo, 0, len(config.Packages))
+	pkgs := make([]*sallyPackage, 0, len(config.Packages))
 	for name, pkg := range config.Packages {
-		handler := newPackageHandler(config, name, pkg)
+		baseURL := config.URL
+		if pkg.URL != "" {
+			// Package-specific override for the base URL.
+			baseURL = pkg.URL
+		}
+		modulePath := path.Join(baseURL, name)
+		docURL := "https://" + path.Join(config.Godoc.Host, modulePath)
+
+		pkg := &sallyPackage{
+			Name:       name,
+			Desc:       pkg.Desc,
+			ModulePath: modulePath,
+			DocURL:     docURL,
+			GitURL:     pkg.Repo,
+		}
```

Comment:
nit: grouping

```suggestion

		var (
			modulePath = path.Join(baseURL, name)
			docURL     = "https://" + path.Join(config.Godoc.Host, modulePath)
			pkg        = &sallyPackage{
				Name:       name,
				Desc:       pkg.Desc,
				ModulePath: modulePath,
				DocURL:     docURL,
				GitURL:     pkg.Repo,
			}
		)
```

Reply:
Same comment as above. The var form is just less readable here. `:=` for non-zero value initialization has always been the way.